pkg/virt-handler/vm.go

```diff
@@ -1182,11 +1184,12 @@ func _guestAgentCommandSubsetSupported(requiredCommands []string, commands []v1.
 
 }
 
-func isGuestAgentSupported(vmi *v1.VirtualMachineInstance, commands []v1.GuestAgentCommandInfo) bool {
+func isGuestAgentSupported(vmi *v1.VirtualMachineInstance, commands []v1.GuestAgentCommandInfo) (bool, string) {
 	log.Log.V(3).Object(vmi).Infof("checking guest agent: %v", commands)
 	if !_guestAgentCommandSubsetSupported(RequiredGuestAgentCommands, commands) {
-		log.Log.V(3).Object(vmi).Info("This guest agent doesn't support required basic commands")
-		return false
+		basicCommandsUnsupportedReason := "This guest agent doesn't support required basic commands"
+		log.Log.V(3).Object(vmi).Info(basicCommandsUnsupportedReason)
+		return false, basicCommandsUnsupportedReason
```

Comment:
I think we could always return the reason and we could log it in the place of the caller. What do you think?

Reply:
Moved it to the caller function, thanks.